tensorflow/core/graph/gpu_fusion_pass.cc

```diff
@@ -421,30 +412,27 @@ bool ROCmFusionPass::RunPass(Graph* graph) {
   }
 
   std::vector<std::unique_ptr<ROCmFusionOpBase> > fusions;
-  std::set<const Node*> fused_nodes;
 
   // Initialize a vector of all the fusion operations we currently support
-  InitializeFusions(fusions, graph, fused_nodes);
-
-  std::vector<Node*> order;
-  GetPostOrder(*graph, &order);  // This will give us reverse topological sort.
-
-  for (const Node* n : order) {
-
-    if (fused_nodes.count(n)) {
-      // We have fused already this node...skip it
-      // Note that we are traversing nodes in reverse topological order, and
-      // matches are found by comparing node sequences from back to front, so
-      // we will hit nodes that we have already fused into a fusion operation.
-      continue;
-    }
-
-    for (auto& fusion : fusions) {
-      if (fusion->DoFusion(n)) {
-        // bail out after the first successful fusion ...
-        // cannot do more than one fusions on a op
-        break;
+  InitializeFusions(fusions, graph);
+
+  for (auto& fusion : fusions) {
+    std::vector<Node*> order;
+    GetPostOrder(*graph,
+                 &order);  // This will give us reverse topological sort.
+
+    std::set<const Node*> fused_nodes;
+
+    for (const Node* n : order) {
+      if (fused_nodes.count(n)) {
```

Comment:
@deven-amd in this new version, wouldn't `fused_nodes` be always empty and this check always be `false`?

Reply:
let me take back my above question.